shared-module/gamepad/__init__.c

```diff
@@ -34,25 +34,49 @@
 
 
 void gamepad_tick(void) {
+    static uint8_t last = 0;
+    uint8_t current = 0;
+    uint8_t bit = 1;
+    digitalio_digitalinout_obj_t* data_pin;
+    digitalio_digitalinout_obj_t* clock_pin;
+    digitalio_digitalinout_obj_t* latch_pin;
+
     gamepad_obj_t* gamepad_singleton = MP_STATE_VM(gamepad_singleton);
     if (!gamepad_singleton) {
         return;
     }
-    uint8_t gamepad_current = 0;
-    uint8_t bit = 1;
-    for (int i = 0; i < 8; ++i) {
-        digitalio_digitalinout_obj_t* pin = gamepad_singleton->pins[i];
-        if (!pin) {
-            break;
+    if (gamepad_singleton->pins[0]) {
```

Comment:
Please use the object's type to determine what to do.

Reply:
Good idea.